web/api/v1/api.go

```diff
@@ -574,6 +585,120 @@ func (api *API) alertmanagers(r *http.Request) (interface{}, *apiError, func())
 	return ams, nil, nil
 }
 
+// AlertDiscovery has info for all active alerts.
+type AlertDiscovery struct {
+	ActiveAlerts []*Alert `json:"activeAlerts"`
+}
+
+// Alert has info for an alert.
+type Alert struct {
+	Labels      labels.Labels `json:"labels"`
+	Annotations labels.Labels `json:"annotations"`
+	State       string        `json:"state"`
+	ActiveAt    *time.Time    `json:"activeAt,omitempty"`
+}
+
+func (api *API) alerts(r *http.Request) (interface{}, *apiError, func()) {
+	alertingRules := api.rulesRetriever.AlertingRules()
+	alerts := []*Alert{}
+
+	for _, alertingRule := range alertingRules {
+		alerts = append(
+			alerts,
+			rulesAlertsToAPIAlerts(alertingRule.ActiveAlerts())...,
+		)
+	}
+
+	res := &AlertDiscovery{ActiveAlerts: alerts}
+
+	return res, nil, nil
+}
+
+func rulesAlertsToAPIAlerts(rulesAlerts []*rules.Alert) []*Alert {
+	apiAlerts := []*Alert{}
+	for _, ruleAlert := range rulesAlerts {
+		apiAlert := &Alert{
+			Labels:      ruleAlert.Labels,
+			Annotations: ruleAlert.Annotations,
+			State:       ruleAlert.State.String(),
+			ActiveAt:    &ruleAlert.ActiveAt,
+		}
+
+		apiAlerts = append(apiAlerts, apiAlert)
+	}
+
+	return apiAlerts
+}
+
+// RuleDiscovery has info for all rules
+type RuleDiscovery struct {
+	RuleGroups []*RuleGroup `json:"groups"`
+}
+
+// RuleGroup has info for rules which are part of a group
+type RuleGroup struct {
+	Name           string          `json:"name"`
+	File           string          `json:"file"`
+	AlertingRules  []alertingRule  `json:"alertingRules"`
+	RecordingRules []recordingRule `json:"recordingRules"`
+	Interval       time.Duration   `json:"interval"`
+}
+
+type alertingRule struct {
+	Name         string        `json:"name"`
+	Query        string        `json:"query"`
+	Duration     time.Duration `json:"duration"`
+	Labels       labels.Labels `json:"labels"`
+	Annotations  labels.Labels `json:"annotations"`
+	ActiveAlerts []*Alert      `json:"activeAlerts"`
```

Comment:
Same as above: Happy to rename this to `Alerts`. `ActiveAlerts` gives the user a hint, that resolved alerts are not included.